extensions/stackdriver/log/logger.h

```diff
@@ -74,18 +88,55 @@ class Logger {
   // Generic method to fill log entry and flush it.
   void fillAndFlushLogEntry(const ::Wasm::Common::RequestInfo& request_info,
                             const ::Wasm::Common::FlatNode& peer_node_info,
+                            bool outbound,
                             google::logging::v2::LogEntry* new_entry);
+
+  // Helper methods to fill destination Labels.
+  void fillDestinationLabels(
+      const ::Wasm::Common::FlatNode& node_info,
+      google::protobuf::Map<std::string, std::string>* label_map);
+
+  // Helper methods to fill source Labels.
+  void fillSourceLabels(
+      const ::Wasm::Common::FlatNode& node_info,
+      google::protobuf::Map<std::string, std::string>* label_map);
+
+  // Helper method to set monitored resource.
+  void setMonitoredResource(
+      const ::Wasm::Common::FlatNode& local_node_info,
+      const std::string& resource_type,
+      google::logging::v2::WriteLogEntriesRequest* log_entries_request);
+
+  // Helper method to initialize inbound/server log entry request.
+  void initializeInboundLogEntryRequest(
+      const flatbuffers::Vector<flatbuffers::Offset<Wasm::Common::KeyVal>>*
+          platform_metadata,
+      const ::Wasm::Common::FlatNode& local_node_info);
+
+  // Helper method to initialize outbound/client log entry request.
+  void initializeOutboundLogEntryRequest(
+      const flatbuffers::Vector<flatbuffers::Offset<Wasm::Common::KeyVal>>*
+          platform_metadata,
+      const ::Wasm::Common::FlatNode& local_node_info);
+
+  // Helper method to get Log Entry Type.
+  Logger::LogEntryType GetLogEntryType(bool outbound) {
+    if (outbound) {
+      return Logger::LogEntryType::Client;
+    }
+    return Logger::LogEntryType::Server;
+  }
+
   // Buffer for WriteLogEntriesRequests that are to be exported.
   std::vector<
       std::unique_ptr<const google::logging::v2::WriteLogEntriesRequest>>
       request_queue_;
 
-  // Request that the new log entry should be written into.
-  std::unique_ptr<google::logging::v2::WriteLogEntriesRequest>
-      log_entries_request_;
-
-  // Estimated size of the current WriteLogEntriesRequest.
-  int size_ = 0;
+  // Stores client/server requests that the new log entry should be written
+  // into.
+  std::unordered_map<Logger::LogEntryType,
```

Comment:
Is it necessary to make a map for only two possible keys? Feel like this is simpler to just have two member variables.

Reply:
it's not just 2 variables.. it will be 4.. request and size.. and then if we add audit logs to the mix it would become 6...

Reply:
Yeah, it makes sense to me to wrap the request and size, but I mean the possible entries for the map is just inbound or outbound. Seems unnecessary to make a map just for two known keys, two member variables would be a better fit here I think. Even it grows to 3 types later, map still seems to be an unnecessary overhead.

Reply:
Well it makes code cleaner, otherwise we will just have too many if else's...checkout the flush methods itself..

This is the current state:
void Logger::flush(Logger::LogEntryType log_entry_type) {
  std::unique_ptr<Logger::WriteLogEntryRequest> request;
  if(log_entry_type == LogEntryType::Client){
    request = client_log_entry_request_;
  } else if(log_entry_type == LogEntryType::Server){
    request = server_log_entry_request_;
  } else {
    request = audit_log_entry_request_;
  }
  std::unique_ptr<google::logging::v2::WriteLogEntriesRequest> cur =
      std::make_unique<google::logging::v2::WriteLogEntriesRequest>();
  cur->set_log_name(request->log_name());
  cur->mutable_resource()->CopyFrom(request->resource());
  *cur->mutable_labels() = request->labels();

  // Swap the new request with the old one and export it.
  request.swap(cur);
  request_queue_.emplace_back(std::move(cur));

  // Reset size counter.
  if(log_entry_type == LogEntryType::Client){
    client_log_entry_request_size_ = 0;
  } else if(log_entry_type == LogEntryType::Server){
    server_log_entry_request_size_ = 0;
  } else {
    audit_log_entry_request_size_ = 0;
  }
}

without map this would look like this:
void Logger::flush(Logger::LogEntryType log_entry_type) {
  std::unique_ptr<Logger::WriteLogEntryRequest> request;
  if(log_entry_type == LogEntryType::Client){
    request = client_log_entry_request_;
  } else if(log_entry_type == LogEntryType::Server){
    request = server_log_entry_request_;
  } else {
    request = audit_log_entry_request_;
  }
  std::unique_ptr<google::logging::v2::WriteLogEntriesRequest> cur =
      std::make_unique<google::logging::v2::WriteLogEntriesRequest>();
  cur->set_log_name(request->log_name());
  cur->mutable_resource()->CopyFrom(request->resource());
  *cur->mutable_labels() = request->labels();

  // Swap the new request with the old one and export it.
  request.swap(cur);
  request_queue_.emplace_back(std::move(cur));

  // Reset size counter.
  if(log_entry_type == LogEntryType::Client){
    client_log_entry_request_size_ = 0;
  } else if(log_entry_type == LogEntryType::Server){
    server_log_entry_request_size_ = 0;
  } else {
    audit_log_entry_request_size_ = 0;
  }
}

The logic looks more clearer as there are going to be more than 2 types of logs and it's less prone to error...

Reply:
These two blocks seem to be identical, but yeah if this makes code simple let's do this way.